unified_planning/io/python_writer.py

```diff
@@ -47,7 +47,7 @@ def walk_forall(self, expression, args):
     def walk_variable_exp(self, expression, args):
         assert len(args) == 0
         v = expression.variable()
-        return f'up.model.Variable("{v.name()}", {_print_python_type(v.type())})'
+        return f'emgr.VariableExp(up.model.Variable("{v.name()}", {_print_python_type(v.type())}))'
```

Comment:
Was this a bug of the python printer? How is it not caught by a test?

Reply:
Yes, it was a bug. 
@Framba-Luca can you answer the question `How is it not caught by a test?`

Reply:
@alvalentini @Framba-Luca  Not sure to follow, is/was this a bug or not? If it was then we need a test that is failing with the previous version and is fixed now, if it was not a bug, that we need justification for this change.

Reply:
Same also for the other changes below